### PR TITLE
fix: Fix nested schema generation

### PR DESCRIPTION
**Pull request by kaavee315**
> [!IMPORTANT]
> Update `remove_json_ref` in `abs.py` to handle nested JSON schema references with `merge_props=True`.
> 
>   - **Behavior**:
>     - Update `remove_json_ref` in `abs.py` to use `merge_props=True` in `jsonref.replace_refs()` to handle nested JSON schema references.
>   - **Misc**:
>     - No other changes or impacts on other parts of the codebase.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=ComposioHQ%2Fcomposio&utm_source=github&utm_medium=referral)<sup> for c75e6bba91554cdf1f80c2bc2298ea58fb1421b8. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: python/composio/tools/base/abs.py [hunk 1]
```diff
@@ -40,6 +40,7 @@ def remove_json_ref(data: t.Dict) -> t.Dict:
             jsonref.replace_refs(
                 obj=data,
                 lazy_load=False,
+                merge_props=True,
```
Comment (shreysingla11) [inline python/composio/tools/base/abs.py:43]: Consider adding a comment explaining why `merge_props=True` is necessary here. This parameter is crucial for handling nested schemas correctly, and documenting this requirement would help future maintainers understand the importance of this setting.

**Comment by shreysingla11** [comment]
I've reviewed the changes and here's my assessment:

The change to add `merge_props=True` to `jsonref.replace_refs()` is a good fix for handling nested schemas. However, I have a few suggestions for improvement:

1. Documentation:
   - The `remove_json_ref` function could benefit from a more detailed docstring explaining its purpose and behavior, especially regarding nested schema handling
   - Consider adding an inline comment explaining why `merge_props=True` is necessary for future maintainers

2. Testing:
   - It would be valuable to add test cases specifically for nested schema scenarios to ensure the fix works as intended
   - Consider testing edge cases where schema merging might be complex

3. Code Quality:
   - The change itself is clean and follows good practices
   - The explicit setting of `merge_props=True` makes the behavior clear and intentional

Overall, this is a good fix that addresses the nested schema generation issue. The changes are minimal and focused, with low risk. I recommend adding the suggested documentation improvements before merging.

Code Quality Rating: 8/10
- Pros: Clean fix, explicit parameter setting, backward compatible
- Areas for improvement: Documentation could be more detailed
